packages/yew-router/src/router.rs

```diff
@@ -87,28 +109,60 @@ pub fn router(props: &RouterProps) -> Html {
     }
 
     html! {
-        <ContextProvider<RouterState> context={(*state).clone()}>
-            {children}
-        </ContextProvider<RouterState>>
+        <ContextProvider<NavigatorContext> context={navi_ctx}>
+            <ContextProvider<LocationContext> context={(*loc_ctx).clone()}>
+                {children}
+            </ContextProvider<LocationContext>>
+        </ContextProvider<NavigatorContext>>
     }
 }
 
 #[derive(Properties, PartialEq, Clone)]
-pub struct BrowserRouterProps {
+pub struct ConcreteRouterProps {
```

Comment:
Missing Rustdoc. This is especially important with function components as they don't show their props struct in their docs